### PR TITLE
call set_proxy_settings during session_impl::setup_listener

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: src/session_impl.cpp [hunk 1]
```diff
@@ -1715,6 +1715,7 @@ namespace aux {
 		}
 
 		ret.udp_sock->set_force_proxy(m_settings.get_bool(settings_pack::force_proxy));
+		ret.udp_sock->set_proxy_settings(proxy());
```
Comment (arvidn) [inline src/session_impl.cpp]: would you mind adding a comment explaining that this has to be called for the ``force_proxy`` settings to take effect? (assuming my understanding is correct). perhaps add more details too if you have a deeper understanding

Comment (aldenml) [inline src/session_impl.cpp]: sure, but let me get the confirmation that this fix the problem first, no need to triggers more CI jobs :)

**Comment by arvidn** [comment]
should I merge this? I expect this patch should go into RC_1_1 as well, right?

**Comment by aldenml** [comment]
I rather wait for confirmation (and the comment), but it could take a while. Your decision.

**Comment by arvidn** [comment]
I was under the impression that @alex-druzhinin had confirmed.

**Comment by aldenml** [comment]
I didn't release any beta, so maybe it was "source code" diff confirmation.

**Comment by aldenml** [comment]
@arvidn added the comment and was confirmed it works

**Pull request merged**